packages/lesswrong/components/hooks/useImageUpload.tsx

```diff
@@ -0,0 +1,303 @@
+import React, { useCallback } from "react";
+import Helmet from "react-helmet";
+import { requireCssVar } from "../../themes/cssVars";
+import {
+  cloudinaryCloudNameSetting,
+  DatabasePublicSetting,
+} from "../../lib/publicSettings";
+import { useTheme } from "../themes/useTheme";
+
+const cloudinaryUploadPresetGridImageSetting = new DatabasePublicSetting<string>(
+  "cloudinary.uploadPresetGridImage",
+  "tz0mgw2s",
+);
+const cloudinaryUploadPresetBannerSetting = new DatabasePublicSetting<string>(
+  "cloudinary.uploadPresetBanner",
+  "navcjwf7",
+);
+const cloudinaryUploadPresetProfileSetting = new DatabasePublicSetting<string | null>(
+  "cloudinary.uploadPresetProfile",
+  null,
+);
+const cloudinaryUploadPresetSocialPreviewSetting = new DatabasePublicSetting<string | null>(
+  "cloudinary.uploadPresetSocialPreview",
+  null,
+);
+const cloudinaryUploadPresetEventImageSetting = new DatabasePublicSetting<string | null>(
+  "cloudinary.uploadPresetEventImage",
+  null,
+);
+const cloudinaryUploadPresetSpotlightSetting = new DatabasePublicSetting<string | null>(
+  "cloudinary.uploadPresetSpotlight",
+  "yjgxmsio",
+);
+
+type CloudinaryImageUploadError = {
+  statusText: string,
+}
+
+type CloundinaryImageUploadResult = {
+  event: "success",
+  info?: {
+    public_id?: string,
+  },
+} | {
+  event:string,
+  info: Record<string, unknown>,
+};
+
+type CloudinaryImageUploadCallback = (
+  error: CloudinaryImageUploadError|null,
+  result: CloundinaryImageUploadResult|null,
+) => void;
+
+type CloudinaryImageUploadFont = {
+  url: string,
+  active: boolean,
+}
+
+type CloudinaryImageUploadFontConfig = Record<string, CloudinaryImageUploadFont|null>;
+
+type CloudinaryImageUploadStyles = {
+  palette?: Record<string, string>,
+  fonts?: CloudinaryImageUploadFontConfig,
+}
+
+declare global {
+  interface Window {
```

Comment:
Seems like the type of thing that should be in a DefinitelyTyped library, but I'm assuming it wasn't

Reply:
Yeah - I had a quick search but couldn't find anything